app/src/main/java/com/asfoundation/wallet/nfts/repository/NFTRepository.kt

```diff
@@ -142,14 +145,14 @@ class NFTRepository @Inject constructor(
       else -> error("Contract not handled")
     }
     val ethGetTransactionCount =
-      web3j.ethGetTransactionCount(fromAddress, DefaultBlockParameterName.LATEST)
-        .sendAsync()
-        .get()
+        web3j.ethGetTransactionCount(fromAddress, DefaultBlockParameterName.LATEST)
+            .sendAsync()
+            .get()
     val nonce = ethGetTransactionCount.transactionCount
-    return RawTransaction.createTransaction(
-      nonce, gasPrice, gasLimit, contractAddress,
-      Numeric.toHexString(data)
-    )
+    val chainID = if (BuildConfig.DEBUG) 4L else 1L
```

Comment:
Maybe add those 1L and 4L chainIds into constants to be easier to read in the future :) Same for the 1_500_000_000L priority fee